### PR TITLE
[feature/cumulative--10.0.0] AB#33495 - Filters Rail Gap Closures

### PR DESCRIPTION
**Pull request by morethanfire**

### CODE REVIEW

File: src/surfaces/filtersRail/__test__/filtersRail.test.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    it('Can render without problems', () => {
+        const wrapper = mountWithTheme(
+            <FiltersRail />,
+        );
+
+        expect(wrapper.exists()).toBe(true);
+    });
+
+    it('Should not render if breakpoint is below md.', () => {
+        window.matchMedia = createMatchMedia(375);
```
Comment (groberts314) [inline src/surfaces/filtersRail/__test__/filtersRail.test.js:39]: Oh, cool!  This helps us simulate different viewport widths?

File: src/surfaces/filtersRail/filtersRailRow.jsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
+import {
+    map,
+} from 'lodash';
+import ClassNames from 'classnames';
+import React, { useState } from 'react';
+import {
+    PROP_TYPES_ROW,
+} from './constants';
+import {
+    BEM_FILTERS_RAIL_ROW, BEM_FILTERS_RAIL_ROW_COMONENT,
```
Comment (groberts314) [inline src/surfaces/filtersRail/filtersRailRow.jsx:10]: This is kinda strange ... these should be on separate lines, right?

File: src/testUtils/enzymeHelpers.jsx [hunk 3]
```diff
@@ -10,6 +12,22 @@ const getThemeProviderWrappingComponent = (customTheme) => ({ children }) => (
...
+    });
+};
+
+export const waitForComponentToPaint = async (wrapper) => {
+    await act(async () => {
+        await new Promise((resolve) => setTimeout(resolve, 0));
+
+        wrapper.update();
+    });
+};
```
Comment (groberts314) [inline src/testUtils/enzymeHelpers.jsx:29]: Very nice.  We might consider adding these in HC (`church-management`) as well so we can use them in tests there too.

**Review by groberts314 — CHANGES_REQUESTED** [review]
Looks pretty good.  I saw a couple of small things and flagged 'em.

File: src/surfaces/drawer/drawerFiltersDrawer.jsx [hunk 4]
```diff
@@ -1,71 +1,46 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import FiltersDrawer from '../../surfaces/filtersDrawer'; // eslint-disable-line import/no-cycle
+import FiltersRail from '../filtersRail';
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply src/surfaces/drawer/drawerFiltersDrawer.jsx:3]: I believe so.

File: src/surfaces/drawer/drawerFiltersDrawer.jsx [hunk 5]
```diff
@@ -1,71 +1,46 @@
...
     style: PropTypes.shape({}),
 };
 
 const defaultProps = {
-    breakpointDown: undefined,
     children: undefined,
     className: undefined,
     id: undefined,
-    rows: undefined,
+    isOpen: undefined,
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply src/surfaces/drawer/drawerFiltersDrawer.jsx:17]: should be `false`, my bad.

File: src/surfaces/filtersRail/__test__/filtersRail.test.js (hunk 1 above)
Comment (morethanfire) [reply src/surfaces/filtersRail/__test__/filtersRail.test.js:39]: Yep, kinda cool.

File: src/surfaces/filtersRail/filtersRailRow.jsx (hunk 2 above)
Comment (morethanfire) [reply src/surfaces/filtersRail/filtersRailRow.jsx:10]: Yes.

File: src/testUtils/enzymeHelpers.jsx (hunk 3 above)
Comment (morethanfire) [reply src/testUtils/enzymeHelpers.jsx:29]: I agree! :D

File: src/surfaces/filtersRail/filtersRailRow.jsx (hunk 2 above)
Comment (morethanfire) [reply src/surfaces/filtersRail/filtersRailRow.jsx]: Fixed.

File: src/surfaces/drawer/drawerFiltersDrawer.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (morethanfire) [reply src/surfaces/drawer/drawerFiltersDrawer.jsx]: Fixed.

File: src/inputs/select/selectCustomDropdownMenu.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+function SelectCustomDropdownMenu() {
+    renderHiddenField(valueArray) {
+        const options = this._visibleOptions = this.filterOptions(this.props.multi && this.props.removeSelected ? valueArray : null);
+        const menu = this.renderMenu(options, valueArray);
+        const styles = Object.assign({}, this.props.menuContainerStyle, {visibility: 'hidden'});
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply src/inputs/select/selectCustomDropdownMenu.jsx]: I cannot event count on one hand how many times i have removed this dang file. It should be removed now though. 🤞

File: docs/src/surfaces/filtersRail/filtersRailDrawerSample.jsx [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+
+        switch (radioItemId) {
+            case RESULT_TYPE_RADIO_ITEM_ACTIVE_ID:
+                newResultTypeIndex = RESULT_TYPE_RADIO_ITEM_ACTIVE_INDEX;
+
+                break;
+            case RESULT_TYPE_RADIO_ITEM_INACTIVE_ID:
+                newResultTypeIndex = RESULT_TYPE_RADIO_ITEM_INACTIVE_INDEX;
+
+                break;
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/surfaces/filtersRail/filtersRailDrawerSample.jsx:143]: I like it the other way. Bwahaha.

[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/surfaces/filtersRail/filtersRailDrawerSample.jsx:143]: .... But I am willing to change. I probably just need to hear your points to be convinced... maybe.

File: docs/src/surfaces/filtersRail/filtersRail.jsx [hunk 8]
```diff
@@ -2,47 +2,101 @@ import {
...
 
-function DocsFiltersRail() {
-    const descriptionCopy = filtersRailDoc.description;
+const propTypes = {
+    location: PropTypes.shape({
+        pathname: PropTypes.string,
+    }).isRequired,
+};
+
+function DocsActionBar(props) {
```
[the remark this answers is not in the document]
Comment (morethanfire) [reply docs/src/surfaces/filtersRail/filtersRail.jsx]: Fixed.

File: docs/src/surfaces/filtersRail/filtersRailDrawerSample.jsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply docs/src/surfaces/filtersRail/filtersRailDrawerSample.jsx:143]: It's more of an aesthetic thing, and difficult to put into words and justify rationally.  Also, not something I feel _that_ passionately about.  Probably just something we wanna stack hands one and try to request everyone does it the same way.  If the majority opinion is not on my said, so be it. :-)

File: docs/src/surfaces/filtersRail/filtersRail.jsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply docs/src/surfaces/filtersRail/filtersRail.jsx]: Looks good.  Resolving.

File: src/inputs/select/selectCustomDropdownMenu.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply src/inputs/select/selectCustomDropdownMenu.jsx]: I'm not seeing this diff any longer, so I'll resolve this thread.

File: src/surfaces/drawer/drawerFiltersDrawer.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (groberts314) [reply src/surfaces/drawer/drawerFiltersDrawer.jsx]: Lookin' good.  Resolving.

File: src/surfaces/filtersRail/filtersRailRow.jsx [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+    map,
+} from 'lodash';
+import ClassNames from 'classnames';
+import React, { useState } from 'react';
+import {
+    PROP_TYPES_ROW,
+} from './constants';
+import {
+    BEM_FILTERS_RAIL_ROW,
+    BEM_FILTERS_RAIL_ROW_COMONENT,
```
Comment (groberts314) [inline src/surfaces/filtersRail/filtersRailRow.jsx:11]: Oh shoot.  I just realized "component" is misspelled in this constant.  It should be `BEM_FILTERS_RAIL_ROW_COMPONENT`.  Sorry I missed this earlier.

File: src/surfaces/filtersRail/filtersRailRow.jsx (hunk 2 above)
Comment (groberts314) [reply src/surfaces/filtersRail/filtersRailRow.jsx]: Looks good.  Resolving.

**Comment by groberts314** [comment]
@morethanfire I think just [this one last thing](https://github.com/saddlebackdev/react-cm-ui/pull/220#discussion_r501865510) and then I will give this my green check mark o' approval.

File: src/surfaces/filtersRail/filtersRailRow.jsx (hunk 9 above)
Comment (morethanfire) [reply src/surfaces/filtersRail/filtersRailRow.jsx]: All fixed up. Thanks!

Comment (groberts314) [reply src/surfaces/filtersRail/filtersRailRow.jsx]: Looks good as of 7aa99cd.  Resolving.

**Review by groberts314 — APPROVED** [review]

**Pull request merged**